gammapy/datasets/core.py

```diff
@@ -456,12 +463,13 @@ def info_table(self, cumulative=False):
         """
         if not self.is_all_same_type:
             raise ValueError("Info table not supported for mixed dataset type.")
+        if self[0].tag == "FluxPointsDataset":
```

Comment:
Looking at these tests, it might be that a `Datasets.type` property returning an `Enum` with something like:
* 0 = `heterogeneous`
* 1 = `MapDataset`
etc. might  be of use.